ls1intum/Artemis
# `Quiz exercises`: Add single choice mode for multiple choice questions

### Checklist
#### General


#### Server
- [ ] N/A. ~~I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).~~
#### Client
- [ ] N/A ~~I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).~~

### Motivation and Context
Closes #3635

### Description
Adds a new option `Single Choice Mode` to multiple choice questions. If enabled it enforces that the question has exactly one correct answer and scoring mode `ALL_OR_NOTHING`.

From the student view it changes the text above the answer options to indicate that only one answer is correct and makes the answer options behave like radio buttons, in that at most one can be selected at a time. It is still possible to select no options to support possible future scoring modes where not answering may be desirable over guessing. Furthermore the scoring info dialogue is adjusted slightly as these questions cannot be answered partially correctly.


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Quiz Exercise with multiple Choice Questions

1. Edit the quiz
2. Play around with the single choice option, the scoring mode and different combinations of correct and wrong answer options
3. Check that the scoring mode always is `ALL_OR_NOTHING` if single choice is activated and that that quiz can only be saved if there is exactly one correct answer for single choice questions.
4. Start & Participate in the Quiz
5. Ensure that it is clearly visible which questions are single choice and which are not. Ensure that for single choice questions at most one answer option can be selected at a time.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


Edit MC Questions:
![image](https://user-images.githubusercontent.com/44437514/161333487-16cbfdce-b751-4053-9714-b228b1be4c93.png)

Answer Single Choice Mode Question:
![image](https://user-images.githubusercontent.com/44437514/161340400-47e0a4fd-b94e-4c16-ad27-dddf1f090f62.png)
